backend/api/endpoints/auth.py

```
@@ -0,0 +1,53 @@
+            "token": "mock-token-123",
+            "user": {
+                "id": user.id,
+                "username": user.username,
+                "role": user.role,
+                "email": user.email
+            }
+        }
+    raise HTTPException(status_code=401, detail="Invalid credentials")
+
+@router.get("/me")
+async def get_current_user(token: str, db: Session = Depends(get_db)):
```

Ruff (0.14.5)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)